project/apps/web/lib/api/auth.ts

```diff
@@ -10,12 +12,42 @@ export interface UserDetails {
   username: string;
 }
 
-export const signUp = (signUpDto: SignUpDto) =>
-  apiCall("post", "/auth/signup", signUpDto);
-
-export const signIn = (signInDto: SignInDto) =>
-  apiCall("post", "/auth/signin", signInDto);
-
-export const signOut = () => apiCall<void>("post", "/auth/signout");
+interface AuthState {
+  user: UserDetails | null;
+  signUp: (signUpDto: SignUpDto) => Promise<void>;
+  signIn: (signInDto: SignInDto) => Promise<void>;
+  signOut: () => Promise<void>;
+  fetchUser: () => Promise<void>;
+  refreshAccessToken: () => Promise<void>;
+}
 
-export const me = (): Promise<UserDetails> => apiCall("get", "/auth/me");
+export const useAuthStore = create<AuthState>()(
+  persist(
+    (set, get) => ({
+      user: null,
+      
+      signUp: async (signUpDto: SignUpDto) => {
+        await apiCall<void>("post", "/auth/signup", signUpDto);
+        await (get() as AuthState).fetchUser();
+      },
+      signIn: async (signInDto: SignInDto) => {
+        await apiCall<void>("post", "/auth/signin", signInDto),
+        await (get() as AuthState).fetchUser();
+      },
```

Comment:
with the fix of signUp and signIn, these endpoints should return u the userData so no need to do another api call to fetchUser